docs: release notes v1.3.2 (#6518)
scalameta/metals: website/blog/2024-06-18-thallium.md

---
author: Katarzyna Marek
title: Metals v1.3.2 - Thallium
authorImageURL: https://github.com/kasiaMarek.png
---

> We're happy to announce the release of Metals v1.3.2, which besides many bugfixes brings improvements to working with non-compiling code.

Reviewer: ```suggestion
We're excited to announce the release of Metals v1.3.2. This new version not only addresses numerous bug fixes but also enhances the experience of working with code that does not compile.
```

<table>
<tbody>
  <tr>
    <td>Commits since last release</td>
    <td align="center">86</td>
  </tr>
  <tr>
    <td>Merged PRs</td>
    <td align="center">81</td>
  </tr>
    <tr>
    <td>Contributors</td>
    <td align="center">13</td>
  </tr>
  <tr>
    <td>Closed issues</td>
    <td align="center">27</td>
  </tr>
  <tr>
    <td>New features</td>
    <td align="center">1</td>
  </tr>
</tbody>
</table>

For full details: [https://github.com/scalameta/metals/milestone/68?closed=1](https://github.com/scalameta/metals/milestone/68?closed=1)

Metals is a language server for Scala that works with VS Code, Vim, Emacs, Zed,
Helix and Sublime Text. Metals is developed at the
[Scala Center](https://scala.epfl.ch/) and [VirtusLab](https://virtuslab.com)
with the help from contributors from the community.

## TL;DR

Check out [https://scalameta.org/metals/](https://scalameta.org/metals/), and
give Metals a try!

> 

Reviewer: ```suggestion
```

- [Outline compilation](#outline-compilation)
- [Use presentation compiler as fallback for references search](#use-presentation-compiler-as-fallback-for-references-search)
- [Allow to override debug server startup timeout](#allow-to-override-debug-server-startup-timeout)

## Outline compilation

> Metals relies on information produced during compilation for many of its functionalities. This limits the usefulness of many Metals features when the code is broken either by providing outdated information or no information if the project was never successfully compiled.

Reviewer: ```suggestion
Metals relies on information produced during compilation for much of its functionality. This limits the usefulness of many Metals features when the code is broken either by providing outdated information or no information if the project was never successfully compiled.
```

Starting with this release Metals will utilize [outline compilation](https://github.com/scala/scala/blob/d800253bbd4cf90d2cf863f1b284dac3561e7446/src/compiler/scala/tools/nsc/settings/ScalaSettings.scala#L347) to get current information about not compiling files for Scala 2.12.x and 2.13.x. This should improve the experience of working with not compiling code.

> There are a few limitations to this improvement, the biggest being that the information from outline compilation will be only available within the same build target (module).

Reviewer: Could also add something about Scala 3 like:

```
Further work is being done on a similar approach for Scala 3, which should be available in one of the upcoming releases.
```

![broken-code](https://i.imgur.com/ZBwWGTK.gif)
*Completions on not compiling code*

## Use presentation compiler as fallback for references search

> Before this release Metals would always use information from SemanticDB to search for references. Since SemanticDB is produced during compilation it would fail to find references for code that didn't compile. Now if SemanticDB information is missing or outdated Metals will fallback to using presentation compiler for reference search.

Reviewer: ```suggestion
Before this release Metals would always use information from SemanticDB to search for references. Since SemanticDB is produced during compilation it would fail to find references for code that didn't compile. Now if SemanticDB information is missing or outdated Metals will fallback to using presentation compiler for reference search for all the files this concerns.
```

![uncompiled-go-to-ref](https://i.imgur.com/tI43rhl.gif)
*Find references on non-compiling code*

## Allow to override debug server startup timeout

> Until this release there was a fixed timeout for debug server startup time set to one minute. This proved to be insufficient for some larger projects and can now be overridden using a server property, e.g. `-Dmetals.debug-server-start-timeout=90` increases the timeout to 90 seconds.

Reviewer: ```suggestion
Until this release there was a fixed timeout for debug server startup time set to one minute. This proved to be insufficient for some larger projects and can now be overridden using a server property, e.g. `-Dmetals.debug-server-start-timeout=90` increases the timeout to 90 seconds.

We will be also working on making the startup time lower, as this heavily impacts the user experience
```
